### ISSUE TITLE
keyword-spacing ignores missing whitespace before "in"

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#7525]
Using ESLint 3.9.1 with  `{ "keyword-spacing": "error" }`

```js
if ("alert"in window) {
    alert("Hi");
}
```

I would expect the above snipped to trigger an error because of the missing whitespace before `in`, but this isn't the case. If I'm misunderstanding the rule, please consider clarifying the documentation.

**Comment by not-an-aardvark** [comment]
Thanks for the report; I can confirm that no error is reported there. However, I wonder if this should be handled by [`space-infix-ops`](http://eslint.org/docs/rules/space-infix-ops) rather than `keyword-spacing`. (At the moment, neither rule reports an error.)

**Comment by mysticatea** [comment]
Thank you for the report.

`space-infix-ops` should warn it.
`keyword-spacing` has the test to check confliction: https://github.com/eslint/eslint/blob/578c373d0357559fb0aedd12d6bad8eb4bb829ff/tests/lib/rules/keyword-spacing.js#L772-L774

**Comment by fasttime** [comment]
Thanks for the replies. `space-infix-ops` seems fine here, I hadn't thought of that rule.

After a quick test, I think `instanceof` shows the same behavior, and neither `space-infix-ops` nor `keyword-spacing` reports an error when a space is missing before the keyword (like in `"foo"instanceof Object`). I haven't checked this with unary ops like `new`, `void`, etc.
